### DISCUSSION TITLE
Pinout Images

### DISCUSSION THREAD

**Discussion by mikrocoder** [SpenceKonde/DxCore#519] [General]
Hi, You would like to have new pinouts for all controllers. My first question would be, what don't you like about the existing ones? Example of the DB series.

**Comment by mikrocoder** [comment]
Hi,
This is a draft. It is made with Excel. This also means that it is not a vector graphic. Has disadvantages when scaling down.
With all 4 sides, however, the graphic will be very, very large. Because of the multiple assignments. I would divide it into 4 parts. 2x horizontal and 2x vertical. To shorten the labels, I would only remove empty fields if the entire column is empty. Otherwise you have shifts between them.
![Pinout Draft](https://github.com/SpenceKonde/DxCore/assets/42818799/517f23e8-7e9f-4eae-9f80-369d347186a7)

**Comment by pcfreak1201** [comment]
Hi!
Yes, all earlier pinouts are generated "by hand" and we need a lot for all the Dx types.
Therfore the idea from @coburnw was to generate them "automatic" and we'd the idea to do this from the DFPs
Here is an example: https://github.com/coburnw/dx-pinouts/blob/main/DA32.svg
But there is still a lot to do by hand - and I struggle with python. But I still like the idea and effort he put in the project...

**Reply by pcfreak1201** [reply]
... and I would prefer a condensed font like https://fonts.google.com/specimen/Roboto+Condensed ...

**Comment by mikrocoder** [comment]
Hi, I don't know anything about Python. The Google Font looks good. 

One question I keep asking myself when creating the drafts is. Does it still make sense to create such pinouts? The multiple assignment is very extensive and basically everything is in the manual in the I/O multiplex table. We are basically just copying you in color.  :-)

**Comment by mikrocoder** [comment]
I continued to try it out. I still have to think about the colors for Portmux. Connecting lines for the empty spaces also need to be added. It all just takes up a lot of space.
![Pinout EA32 Draft left](https://github.com/SpenceKonde/DxCore/assets/42818799/ef8aa075-2dbc-40ea-bb80-baeda21c8111)
![Pinout EA32 Draft right](https://github.com/SpenceKonde/DxCore/assets/42818799/63883732-db59-4537-bb22-3341ceeffeb9)

**Comment by coburnw** [comment]
As a hardware guy, i certainly appreciate the comment on how the visual data sheets are just copying the assignment chart in color.  But when done right, they sure are pretty.  I saw Tengo10's pinout generator and i couldnt resist that sirens call either.

Spence's comment [here](https://github.com/SpenceKonde/DxCore/discussions/459#discussioncomment-6510777) fully outlines both his wants and potential pitfalls.  If i had spent the time trying to fully understand what he was getting at, i may have had more progress to show with less loss of blood.  Here are the important details of that comment as i see them now:

- The pinouts that Spence wants are supporting documents for HIS core.  His function names dont necessarily correlate with those of the atdf's.
- The atdf files dont contain the physical pin mappings for all possible combinations.
- There are errors in the atdf.
- He is willing to generate parsable files in any format (except yaml) for someone to build his diagrams with.

If i were to come at this again, i would abandon parsing the atdf files and rely solely on the data that Spence supplies.  See pinsdx.txt found [here](https://github.com/SpenceKonde/DxCore/issues/264#issuecomment-1681068341).

I would be happy to support anyone who wanted to make use of pinout-overview-lib, or for that matter the atdf parser.  Tengo10's graphics which the library is derived from are in my mind the cleanest of them all.  Someone with a good eye for color and negative space could make diagrams that really  pop.

**Comment by pcfreak1201** [comment]
Well, from a user-perspective (and hardware-developer) I don't want a universal pinout, but a "good" pinout ;-) Therefore I still do them by hand in Inkscape like here: https://github.com/pcfreak1201/Arduino-Nano-Clone-4.0/blob/main/images/ED-m4808.png
But whenever you see any board, compatible with Spence's core(s), the pinout is always included/referenced.
I think, we need a step in between: e.g a .json-file, generated per chip from the atdf, but human readable and editable / correctable as input for the pinout generator. This would be maintainable inside the core and then the pinout generation should work like a make-file.
Like mikrocoder I'm absolute no python user, but I understand that it is the best way to do it.

**Comment by mikrocoder** [comment]
Hi, In the meantime, I used PowerPoint instead of Excel this time. From this you can save in .svg. It scales nicely and during the creation you have magnetic effects to position the "bricks". I just don't know how small you should make the font so that you can read everything normally. The current font is Roboto Condensed size 12. It's not finished yet, it's an in-progress version.
![AVRxEA28](https://github.com/SpenceKonde/DxCore/assets/42818799/a3c5d4bc-3931-4b79-be03-6d272316745b)
[AVRxEA28_svg.zip](https://github.com/SpenceKonde/DxCore/files/14397648/AVRxEA28_svg.zip)

**Comment by mikrocoder** [comment]
Hello,

I have continued to work on it. This is the current draft. I don't know yet whether the gaps (currently white) should be removed. If you move everything together, the formatting gets lost. I don't know which is better. Just let me know what you think.
![AVRxEA28](https://github.com/SpenceKonde/DxCore/assets/42818799/59453b53-5491-46d7-b8c3-1c0366869119)
[AVRxEA28.zip](https://github.com/SpenceKonde/DxCore/files/15286580/AVRxEA28.zip)
@SpenceKonde , Your opinion is also asked for in the thread, you want it. :-) What did you use to create the previous pinouts? I do it with PowerPoint. It's all manual work and prone to errors, as I've just discovered again. But it looks nice. :-)
